include/grpc/byte_buffer.h

```diff
@@ -34,10 +34,17 @@
 #ifndef GRPC_BYTE_BUFFER_H
 #define GRPC_BYTE_BUFFER_H
 
-#include <grpc/grpc.h>
 #include <grpc/support/slice_buffer.h>
 
-typedef enum { GRPC_BB_SLICE_BUFFER } grpc_byte_buffer_type;
+typedef enum {
+  GRPC_BB_SLICE_BUFFER,
+
+  /* Keep the GRPC_BB_COMPRESSED_* entries in the same order as the
+   * grpc_compression_algorithm enum entries. */
+  GRPC_BB_COMPRESSED_NONE, /* for overriding otherwise compressed channels */
+  GRPC_BB_COMPRESSED_DEFLATE,
```

Comment:
I wonder about splitting these out into an orthogonal property. Something like:

```c
typedef enum {
  GRPC_BB_RAW,
  /* in the future I expect GRPC_BB_PROTOBUF or similar */
} grpc_byte_buffer_type;

struct grpc_byte_buffer {
  grpc_byte_buffer_type type;
  union {
    struct {
      grpc_compression_algorithm compression;
      grpc_slice_buffer slice_buffer;
    } raw;
  };
};
```

Pros:
- it's a little easier to interpret 'type'
- it keeps some competing concerns separated
  - it's easier to extend compression_type and byte_buffer_type independently
- it'll be easier to add a GRPC_BB_PROTOBUF when we need to

Cons:
- it's a larger change
- the structure of byte_buffer is slightly more complex